add markers to video player

remo7k opened the pull request.

ryanvarley reviewed (commented):
I guess the main question is do we want this as an intenral release branch for now or putting it out into the world?

remo7k commented:
Hey @ryanvarley, we can release it to the world, changes made are valid UX improvements around segments

 I've refactored the code and fixed the tests. Please confirm it works as expected
